src/betterproto/__init__.py

```diff
@@ -1868,6 +1868,23 @@ def _validate_field_groups(cls, values):
 
 Message.__annotations__ = {}  # HACK to avoid typing.get_type_hints breaking :)
 
+# monkey patch (de-)serialization functions of class `Message`
+# with functions from `betterproto-rust-codec` if available
+try:
+    import betterproto_rust_codec
+
+    def __parse_patch(self: T, data: bytes) -> T:
+        betterproto_rust_codec.deserialize(self, data)
+        return self
+
+    def __bytes_patch(self) -> bytes:
+        return betterproto_rust_codec.serialize(self)
+
+    Message.parse = __parse_patch
+    Message.__bytes__ = __bytes_patch
+except ModuleNotFoundError:
+    pass
```

Comment:
I'd personally just do something along the lines of
```py
def parse():...
def __bytes__():...
if importlib.util.find_spec("betterproto_rust_codec"):
    import betterproto_rust_codec
	parse = betterproto_rust_codec.deserialize  # edit this so it returns self
	__bytes__ = betterproto_rust_codec.serialize
```

Reply:
I don't really understand why, but directly assigning `__bytes__ = betterproto_rust_codec.serialize` does not work for some reason. So editing `deserialize` would probably don't help either.

What do you think about this approach?

```python3

class Message:
    if HAS_ACCELERATOR:
        def parse(self, data):
            ...
    else:
        def parse(self, data):
            ...
```

I think this might be even more transparent.

Reply:
I find that problematic because it's another level of indentation which is annoying especially with black. If the function isn't binding as a descriptor just leave it as is